e2e_tests/integration/fullscreen.spec.ts

```diff
@@ -0,0 +1,170 @@
+
+    // we can leave fullscreen
+    cy.get(EditorArea).type('{esc}')
+    cy.get(EditorArea)
+      .invoke('height')
+      .should('equal', 286)
+
+    cy.get(EditorArea)
+      .invoke('width')
+      .should('equal', viewportWidth - 80) // normal editor width
+  })
+
+  it('can fullscreen main editor by clicking fullscreen icon', () => {
+    const { viewportWidth, viewportHeight } = Cypress.config()
+
+    cy.get(EditorArea).type('{shift}\n'.repeat(20))
+    // Should stop at 286 px (13 lines height)
+    cy.get(EditorArea)
+      .invoke('height')
+      .should('equal', 286)
+
+    cy.get(EditorArea)
+      .invoke('width')
+      .should('equal', viewportWidth - 80) // normal editor width
+
+    cy.get('[data-testid="editor-fullscreen"]').click()
+
+    cy.get(EditorArea)
+      .invoke('height')
+      .should('equal', viewportHeight)
+
+    cy.get(EditorArea)
+      .invoke('width')
+      .should('equal', viewportWidth) // editor is now full width
+
+    cy.get('[data-testid="editor-fullscreen"]').click()
+    cy.get(EditorArea)
+      .invoke('height')
+      .should('equal', 286)
+
+    cy.get(EditorArea)
+      .invoke('width')
+      .should('equal', viewportWidth - 80) // normal editor width
+  })
+
+  it('re-usable fram can also use fullscreen', () => {
+    const { viewportWidth, viewportHeight } = Cypress.config()
+    cy.executeCommand(':clear')
+    cy.executeCommand('return 1')
+    cy.get('[data-testid="frameCommand"]')
+      .contains('return 1')
+      .click()
+
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .type('{shift}\n'.repeat(20))
+
+    // Should stop at 276 px (max reusable frame height)
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .invoke('height')
+      .should('equal', 276)
+
+    // frame height should stop at 580
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .closest('article')
+      .invoke('height')
+      .should('equal', 580)
+
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .invoke('width')
+      .should('equal', viewportWidth - 184) // normal reusable-editor width
+
+    // goto fullscreen
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .type('{esc}')
+
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .invoke('height')
+      .should('equal', 276) // reusable keeps the same height on fullscren but content take up whole screen
+
+    // frame height is full height of screen
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .closest('article')
+      .invoke('height')
+      .should('equal', viewportHeight - 3) // excluding some margin
+
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .invoke('width')
+      .should('equal', viewportWidth - 94) // reusable editor is not quite full with
+
+    cy.get('[title="Close fullscreen"]').click()
+
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .invoke('height')
+      .should('equal', 276)
+
+    cy.get('[id^=monaco-]')
+      .eq(1)
+      .invoke('width')
+      .should('equal', viewportWidth - 184) // normal editor width
```

Comment:
Can we at least move these numbers to constants with somewhat descriptive names? This would both make it easier to understand, and to change in the future.

Reply:
WIll do!